### ISSUE TITLE
specify snakelike version during installation?

### ISSUE THREAD

**Issue by ceesu** [snakemake-workflows/rna-seq-star-deseq2#71]
Instructions for installation currently start with

`mamba create -c conda-forge -c bioconda --name snakemake snakemake snakedeploy`

However, it seems like with the latest version of snakemake (8.2.3), running snakemake leads to the error 

```
NotImplementedError: Remote providers have been replaced by Snakemake storage plugins. Please use the corresponding storage plugin instead (snakemake-storage-plugin-*).
```

**Comment by Funkoverload** [comment]
Same issue here. When I downgrade to snakemake 7.32.4, I get another errror: 

  File "/extra/sra/.snakemake/conda/6ced698129ee201eb5ebbc98e1ca2dab_/lib/python3.12/sqlite3/__init__.py", line 70, in __getattr__
    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
AttributeError: module 'sqlite3' has no attribute 'OptimizedUnicode'

**Comment by dlaehnemann** [comment]
> Instructions for installation currently start with
> 
> `mamba create -c conda-forge -c bioconda --name snakemake snakemake snakedeploy`
> 
> However, it seems like with the latest version of snakemake (8.2.3), running snakemake leads to the error
> 
> ```
> NotImplementedError: Remote providers have been replaced by Snakemake storage plugins. Please use the corresponding storage plugin instead (snakemake-storage-plugin-*).
> ```

How exactly are you invoking the workflow? This sounds like it might be an issue with using a cluster system. For infos on moving from snakemake 7 to snakemake 8, have a look here (especially at the info on storage providers):
https://snakemake.readthedocs.io/en/latest/getting_started/migration.html#migrating-to-snakemake-8

**Comment by dlaehnemann** [comment]
> Same issue here. When I downgrade to snakemake 7.32.4, I get another errror:
> 
> File "/extra/sra/.snakemake/conda/6ced698129ee201eb5ebbc98e1ca2dab_/lib/python3.12/sqlite3/**init**.py", line 70, in **getattr** raise AttributeError(f"module {**name**!r} has no attribute {name!r}") AttributeError: module 'sqlite3' has no attribute 'OptimizedUnicode'

The `7.32.4` issue is probably an incompatibility with `python3.12`, see here for the deprecation notice of the above attribute:
https://github.com/python/cpython/issues/92547#issue-1229423185

I remember that some `python3.12` incompatibilities should be fixed in snakemake 8, so this might go away once you get the workflow running with snakemake 8. Otherwise, a quick fix can be to downgrade the version of python used manually, to get `sqlite3` to still have the respective attribute:
```
mamba activate snakemake
mamba install python=3.10
```

**Comment by ceesu** [comment]
Thanks so much @dlaehnemann. I got it to work by deleting everything and using the old version of snakemake:

`mamba create -c conda-forge -c bioconda -n snakemake_novasenta_0013 snakemake=7.32.4 `

I also found that using any version of snakemake older than 7.3.4 also causes an f string error, seems to be related to [snakemake/snakemake #2480](https://github.com/snakemake/snakemake/issues/2480#issuecomment-1882770815)

@Funkoverload I also that error with version 2.0.0 of this pipeline, but seems that it was fixed most recent version  by #69 . maybe you can try installing with 
```
snakedeploy deploy-workflow https://github.com/snakemake-workflows/rna-seq-star-deseq2 . 
```

Instead of what's given  in the instructions, i.e. ```snakedeploy deploy-workflow https://github.com/snakemake-workflows/rna-seq-star-deseq2 . --tag v2.0.0```

**Comment by dlaehnemann** [comment]
Ah, yes. I have already forgotten about #69 and the similar error there. I have now created a new release, so that the following should also pull in the latest changes (and provide them via a stable tag version):
```
snakedeploy deploy-workflow https://github.com/snakemake-workflows/rna-seq-star-deseq2 . --tag v2.1.0
```

It should take a day for the command update in the snakemake workflow catalog, but with the release already done, the command I provide here should already work.

I'll close this issue for now. Feel free to reopen if any of this re-occurs or open new issues for new problems.
